pvlib/solarposition.py

```diff
@@ -409,36 +416,32 @@ def get_sun_rise_set_transit(time, latitude, longitude, how='numpy',
     lat = latitude
     lon = longitude
 
-    if not isinstance(time, pd.DatetimeIndex):
-        try:
-            time = pd.DatetimeIndex(time)
-        except (TypeError, ValueError):
-            time = pd.DatetimeIndex([time, ])
+    # times must be localized
+    if not times.tz:
+        raise ValueError('sun_rise_set_ephem: times must be localized')
```

Comment:
IMO a more Pythonic way would be to get `tz` here which will raise an `AttributeError`, then use `tz` obtained below on lines 436, 438, and 440.

Or if you want a more specific error message, then use duck typing

```python
try:
    tzinfo = times.tz
except AttributeError:
    raise ValueError('sun_rise_set_ephem: times must be localized')
```

Then below ...
```python
transit = pd.to_datetime(transit*1e9, unit='ns', utc=True).tz_convert(
    tzinfo).tolist()
sunrise = pd.to_datetime(sunrise*1e9, unit='ns', utc=True).tz_convert(
    tzinfo).tolist()
sunset = pd.to_datetime(sunset*1e9, unit='ns', utc=True).tz_convert(
    tzinfo).tolist()
```

See the [Pythonic documentation on duck typing and eafp](https://docs.python.org/3/glossary.html) for explaining for this style

Reply:
Well it is a good idea but `tz = NoneType` doesn't trigger an attribute error.

Reply:
Sorry I was wrong about `AttributeError`, looks like times always has a `tz` attribute, although, wouldn't it be okay if it were `None`? That would mean localized, but naive, which should work fine.

Reply:
With naïve datetimes, the code will run but not give the expected result, except at locations with UTC timezone.